packages/plugin-federation/src/schema-builder.ts

```diff
@@ -64,18 +79,7 @@ schemaBuilderProto.toSubGraphSchema = function toSubGraphSchema({
           args: {
             url: linkUrl,
             import: [
-              '@key',
-              '@shareable',
-              '@inaccessible',
-              '@tag',
-              '@provides',
-              '@requires',
-              '@external',
-              '@extends',
-              '@override',
-              '@authenticated',
-              '@policy',
-              '@requiresScopes',
+              ...federationDirectives,
               options.composeDirectives ? '@composeDirective' : null,
               hasInterfaceObjects ? '@interfaceObject' : null,
```

Comment:
I think potentially it might make more sense to have `authenticated`, `requiresScopes`, and `policy`, and maybe even all the directives work like `composeDirective` and `interfaceObject` where they are only imported when used.  I think being able to explicitly add your own directives is good, but I think we should potentially just auto-import any directive that gets used through a plugin feature

Reply:
Sure, makes sense - is there a simple way to derive at this point which of these directives have been used, either via (e.g.) the `requiresScopes` option or the directives array on each query/ mutation/ enum/ etc...?

Reply:
Not yet, I can try to find some time in the next few days to add something like that.  I think it would probably be easiest to track that as the directives are added into the schema in the plugin class